test/type_traits.cpp

```diff
@@ -759,6 +975,153 @@ TEST_CASE("underlying_type_t", "[meta.trans.other]") {
                      std::underlying_type<E>::type>::value);
 }
 
+// template <class Fn, class... ArgTypes> struct invoke_result;
+template <typename T, typename Enable = void>
+struct no_result : std::true_type {};
+
+template <typename T>
+struct no_result<T,
+                 typename std::conditional<false, typename T::type, void>::type>
```

Comment:
`slb::void_t<typename T::type>`?

Reply:
I see that we already have a `has_nested_type` below, but I guess
we're using this here because the `void_t` isn't tested at this point 😞

Reply:
I don't want to have unit tests depending on other units also being tested, at least not just for convenience of writing. I will introduce a `no_result_void` for clarity.

Reply:
Yep, understood.